core/services/beacon/settings.py

```diff
@@ -0,0 +1,117 @@
+import json
+import re
+import socket
+from typing import Any, Dict
+
+import psutil
+from commonwealth.settings import settings
+from pykson import (
+    IntegerField,
+    JsonObject,
+    ListField,
+    MultipleChoiceStringField,
+    ObjectListField,
+    StringField,
+)
+
+
+class InvalidIpAddress(Exception):
+    pass
+
+
+class DefaultSettings(JsonObject):
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+
+class Interface(JsonObject):
+    name = StringField()
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+    def get_phys(self) -> psutil._common.snicaddr:
+        """
+        return Physical interface from psutil
+        """
+        return psutil.net_if_addrs()[self.name]
```

Comment:
You can use [AF_INET](https://docs.python.org/3/library/socket.html#socket.AF_INET) as `family` argument to avoid the mac address and IPv6.